sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/AsynchronousMetricStorage.java

```diff
@@ -39,17 +39,25 @@
   private static final Logger logger = Logger.getLogger(AsynchronousMetricStorage.class.getName());
 
   private final ThrottlingLogger throttlingLogger = new ThrottlingLogger(logger);
+  private final RegisteredReader registeredReader;
   private final MetricDescriptor metricDescriptor;
+  private final AggregationTemporality aggregationTemporality;
   private final TemporalMetricStorage<T, U> metricStorage;
   private final Aggregator<T, U> aggregator;
   private final AttributesProcessor attributesProcessor;
   private Map<Attributes, T> accumulations = new HashMap<>();
 
   private AsynchronousMetricStorage(
+      RegisteredReader registeredReader,
       MetricDescriptor metricDescriptor,
       Aggregator<T, U> aggregator,
       AttributesProcessor attributesProcessor) {
+    this.registeredReader = registeredReader;
     this.metricDescriptor = metricDescriptor;
+    this.aggregationTemporality =
+        registeredReader
+            .getReader()
+            .getAggregationTemporality(metricDescriptor.getSourceInstrument().getType());
```

Comment:
A small optimization to calculate the temporality one instead of each time a collection occurs.